zstacklib/zstacklib/network/ovs/utils.py

```
@@ -0,0 +1,312 @@
+    Args:
+        bdf_str: BDF string like '0000:65:00.1'.
+
+    Raises:
+        OvsError: If format is invalid.
+    """
+    pattern = re.compile(r'\d{4}(:[0-9a-fA-F]{2}){2}.\d$')
+    ret = re.match(pattern, bdf_str)
+    if ret is not None:
+        if ret.span()[1] == 12:
+            return
+    raise OvsError(f'BDF format error. bdf:{bdf_str}')
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)